### PR TITLE
Refine CUDA CMake options and usage in scripts.

### PR DESCRIPTION
**Pull request by ScottTodd**
Fixes https://github.com/iree-org/iree/issues/11948

* Stop setting CUDA CMake options (`-DIREE_HAL_DRIVER_CUDA=ON -DIREE_TARGET_BACKEND_CUDA=ON`) explicitly in scripts, since these now default to enabled on supported platforms
* Skip downloading our own copy of the CUDA Toolkit if `find_package(CUDAToolkit)` is successful at finding a version somewhere on the system. Note: the user may have a different version than what we automatically download. We can include version information in the `find_package()` call as needed, see https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html.

### CODE REVIEW

**Comment by benvanik** [comment]
+1 to only doing the download stuff if find_package fails

**Comment by ScottTodd** [comment]
> +1 to only doing the download stuff if find_package fails

It looks like before https://github.com/iree-org/iree/pull/11651 we had 
```cmake
if(IREE_TARGET_BACKEND_CUDA OR IREE_HAL_DRIVER_CUDA)
  find_package(CUDAToolkit)

  if(NOT CUDAToolkit_FOUND)
    ...
```

I'll iterate on this a bit.

File: build_tools/third_party/cuda/CMakeLists.txt [hunk 1]
```diff
@@ -91,17 +93,23 @@ else()
...
-    # Download the CUDA toolkit if needed.
-    fetch_cuda_toolkit()
-    if(CUDAToolkit_ROOT)
-      # Download succeeded.
-      message(STATUS "Using downloaded CUDA toolkit: ${CUDAToolkit_ROOT}")
-      set(CUDAToolkit_ROOT "${CUDAToolkit_ROOT}" PARENT_SCOPE)
-      # For some reason having the BIN_DIR set wrong can cause mayhem. Just make
-      # sure it is right.
-      set(CUDAToolkit_BIN_DIR "${CUDAToolkit_ROOT}/bin" PARENT_SCOPE)
+    if(CUDAToolkit_FOUND)
```
Comment (GMNGeoffrey) [inline build_tools/third_party/cuda/CMakeLists.txt:96]: It seems a bit weird that we have this same if-else condition in the main cmake file and here. Can we look for/download cuda all in the same place?

Comment (ScottTodd) [reply build_tools/third_party/cuda/CMakeLists.txt:96]: We _could_, but this is currently set up to first check if CUDA is available and set the option defaults, then only actually download CUDA if the options are set.

```python
# ----------------------------------------------------------
# CMakeLists.txt

check_for_cuda()

if cuda available:
  default = on
else:
  default = off

option(cuda_compiler default)
option(cuda_runtime default)

if cuda_compiler or cuda_runtime:
  setup_cuda()

# ----------------------------------------------------------
# build_tools/third_party/cuda/CMakeLists.txt

setup_cuda():
  download_if_needed()
  # define a few variables
  # add a few libraries
```

**Comment by GMNGeoffrey** [comment]
Ok fair

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**